storage-provider/common/src/deadline.rs

```diff
@@ -217,4 +237,70 @@ impl Deadline {
 
         Ok(())
     }
+
+    async fn generate_proofs_for_partitions<SectorStorage>(
+        &self,
+        partitions: Vec<(PartitionNumber, PartitionState)>,
+        prover_id: [u8; 32],
+        randomness: [u8; 32],
+        post_params: Arc<PoStParameters>,
+        sector_storage: SectorStorage,
+    ) -> Result<(Vec<PartitionNumber>, Vec<PoStProof>), DeadlineError>
+    where
+        SectorStorage: Fn(SectorNumber) -> Option<ProvenSector>,
+    {
+        // Get replicas for sectors part of the partitions
+        let replicas = partitions
+            .iter()
+            .flat_map(|(_id, state)| {
+                state.sectors.iter().map(|sector_number| {
+                    sector_storage(*sector_number)
+                        .ok_or(DeadlineError::SectorNotFound(*sector_number))
+                        .map(|sector| ReplicaInfo {
+                            sector_id: sector.sector_number,
+                            comm_r: sector.comm_r.raw(),
+                            cache_path: sector.cache_path.clone(),
+                            replica_path: sector.sealed_path.clone(),
+                        })
+                })
+            })
+            .collect::<Result<Vec<_>, DeadlineError>>()?;
+
+        // Generate proofs
+        let proofs: Vec<BlstrsProof> = {
+            let post_params = post_params.clone();
+            let post_proof = self.post_proof;
+
+            tokio::task::spawn_blocking(move || {
+                match_post_proof!(
+                    post_proof,
+                    generate_window_post::<_>(
+                        post_proof,
+                        &post_params,
+                        randomness,
+                        prover_id,
+                        replicas
+                    )
+                )
+            })
+        }
+        .await??;
+
+        // Map proofs to our internal type
+        let proofs = proofs
+            .into_iter()
+            .map(|p| PoStProof {
+                post_proof: self.post_proof,
+                proof_bytes: codec::Encode::encode(
+                    &TryInto::<SubstrateProof>::try_into(p.clone()).expect(
+                        "converstion between rust-fil-proofs and polka-storage-proofs to work",
```

Comment:
The expect doesn't show up in the error message so the text feels kinda weird without it
```suggestion
                        "converstion between rust-fil-proofs and polka-storage-proofs should work",
```